### DISCUSSION TITLE
Message format translation in an OpenAI-like API.

### DISCUSSION THREAD

**Discussion by zekrismh** [vllm-project/vllm#1483] [Q&A]
Hello everyone,

I hope you're all doing well. I wanted to reach out regarding the implementation of the Vllm technology, specifically in the context of the "OpenAI Like" feature. I'm currently exploring how Vllm functions when applied to OpenAI-like tasks.  I'm particularly interested in understanding if VLLM can manage the translation from OpenAI message format to, for example, LLAMA message format. **Example**: 
**LLama format :** 
```
prompt = """\
[INST] Hi! [/INST]
Hello! How are you?
[INST]  I'm great, thanks for asking. Could you help me with a task? [/INST]
Of course, I'd be happy to help! Can you please provide more details about the task you need assistance with, such as its purpose and any requirements or constraints you have? This will help me better understand how I can assist you. Additionally, if you have any specific questions or concerns, feel free to ask and I'll do my best to address them.
[INST] How much wood could a wood chuck chuck or something like that? [/INST]
"""
```
**openAI format** 
`prompt = [{"role":"assistant","content":"Hi"},{"role":"user","content":"Hello! How are you?"},{"role":"assistant","content":"I'm great, thanks for asking. Could you help me with a task? "}]`

**Comment by Tostino** [comment]
Right now it uses FastChat to hopefully find the right format. 

I have a PR open to use the HF chat templates: https://github.com/vllm-project/vllm/pull/1493
